pkg/utils/environment/kernel_symbols.go

```diff
@@ -2,303 +2,223 @@ package environment
 
 import (
 	"bufio"
-	"fmt"
+	"io"
 	"os"
 	"strconv"
 	"strings"
 	"sync"
-)
 
-const (
-	kallsymsPath = "/proc/kallsyms"
-	chanBuffer   = 112800 // TODO: check if we really need this buffer size
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/utils"
 )
 
-type KernelSymbol struct {
-	Name    string
-	Type    string
-	Address uint64
-	Owner   string
-}
-type nameAndOwner struct {
-	name  string
-	owner string
+// Kernel symbols do not have an associated size, so we define a sensible size
+// limit to prevent unrelated symbols from being returned for an address lookup
+const maxSymbolSize = 0x100000
+
+var ownersMu sync.RWMutex
```

Comment:
The following owner tracking data structures should be localized to the struct I think. Since this package doesn't expose a singleton (which it certainly could given that a kernel symbol table is a single system file) I don't think we should mix global and struct variables.
That is unless you want to make this a singleton.

Reply:
What purpose does the index:owner mapping serve? I see that when you parse the file you do string->int and when you call `Owner()` you do int->string (via array), but I don't see where the int plays a role in itself. Why not just use strings overall?
Further, if you expect that an array of owner string will be rather small (up to 5 lets say), then the map duplication is redundant (consider that in event args we prefer arrays to a map), and you could also drop the mutex.

Reply:
The purpose is to reduce memory usage, storing a uint16 takes less memory than an owner string. On my machine there are 103 possible owners so I'm not sure it would be smart to drop the map.

Reply:
Maybe it wouldn't be that bad storing the name, on my machine 30622/230472 symbols are owned by modules, and the system symbols could have a memory efficient representation (like a nil)

Reply:
> The purpose is to reduce memory usage, storing a uint16 takes less memory than an owner string. On my machine there are 103 possible owners so I'm not sure it would be smart to drop the map.

That's why I said it depends on the expected count.

> Maybe it wouldn't be that bad storing the name, on my machine 30622/230472 symbols are owned by modules, and the system symbols could have a memory efficient representation (like a nil)

That does sound like a lot and storing the string directly would be costly... So going with the int was a good call.

I still don't like the mixed approach but maybe that means the kernelsymbol table should be a singleton. Tentative on this for now.